cityofaustin/atd-vz-data
# Update CrashEditCoordsMap

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/15867

**Please review these PRs in this order: https://github.com/cityofaustin/atd-vz-data/pull/1398, https://github.com/cityofaustin/atd-vz-data/pull/1399, https://github.com/cityofaustin/atd-vz-data/pull/1400**

This one reimplements the `CrashEditCoords` with `react-map-gl` v7 and prevents map zoom from happening on page scroll.

## Testing
**URL to test:** 
Local

Steps to test:

1. FYI you will need to run `npm i` in the editor folder when switching back and forth between this branch and others without the map library update
2. Go to a crash page and then click on the **Edit Coordinates** button to show the crash location edit map
3. See that there are zoom and fullscreen controls in the top left just like in staging and production
4. ~Try scrolling while hovering the mouse over the crash map. You should see a prompt to use a cooperative gesture to zoom the map in and out.~ This is outdated. It should zoom on scroll.
5. Drag the map, and you should see the marker animations happening just like they do in staging and production. Also notice that the values in the latitude/longitude inputs in the form below the map update when the marker is moved.
6. After moving the marker, click the **Reset** button and the marker and map should jump to the original location of the crash before you started editing.
7. Click the **Cancel** button and the edit map should close and not save a new position.
8. Open the edit map again, move the marker, and then click **Save**. You should see that all three buttons now disable while the update request is happening. You might have to throttle your network speed in browser to see this in action. The edit map closes after the update succeeds.
9. Verify that the location updated by refreshing the page and checking it in the crash map.

---
#### Ship list
